### PR TITLE
Add the /lfx-url command to record LFX program URLs

### PR DESCRIPTION
**Pull request by nate-double-u**
Adds `/lfx-url <url>`, the last automated step of the LFX Mentorship proposal intake (§4.3.5 / §4.5 of #1883).

After a term's export PR is merged, a CNCF admin comments `/lfx-url <url>` on an exported proposal issue. The bot:
- records the URL on the issue (the durable source of truth the export reads back),
- fills it into that term's `lfx-export.json`, `README.md`, and `lfx-tracking.csv` via an accumulating per-term "LFX URLs — <term>" PR, and
- advances the board card to "Posted to LFX".

Guards: restricted to global approvers (same allowlist as `/cncf-approve`); the URL must be an exact program URL (`https://mentorship.lfx.linuxfoundation.org/project/<uuid>`); the issue must be open and its export merged to `main`, else it's rejected atomically with the exact command to re-run; a closed issue is rejected; and the board advance never overrides a manual placement.

Also refactors the export's README/CSV/term-path rendering into shared, unit-tested libs so the export and `/lfx-url` stay in lockstep.

Built and validated end-to-end on the dev fork (nate-double-u/mentoring): full command matrix + the merge-order and no-regress edge cases. 304 unit tests pass.

### CODE REVIEW

File: .github/workflows/lfx-proposal-approvals.yml [hunk 1]
```diff
@@ -353,6 +367,117 @@ jobs:
...
+              // recording now would move the board and leave the files unsynced.
+              // Reject atomically instead — nothing recorded, no board move — and
+              // hand back the exact command to re-run after the export PR merges.
+              // decision.url already passed the strict format gate, so it is safe
+              // to echo here.
+              const lfxTerm = (parseIssueForm(issueBody)['Term'] || '').trim();
+              const { outDir, year, termDir, readmeTitle } = termPaths(lfxTerm);
+              const jsonPath = `${outDir}/lfx-export.json`;
+              let data = null;
+              try {
```
[the remark this answers is not in the document]
Comment (nate-double-u) [reply .github/workflows/lfx-proposal-approvals.yml:423]: Valid finding; tracking it as a follow-up rather than blocking this PR.

The guard derives the term directory from the issue's `Term` field, which is editable. If the Term is changed after the export merges, the guard looks in the wrong directory and **rejects** a legitimate `/lfx-url`. This is fail-safe: an issue number belongs to exactly one term's export, so a wrong Term can only cause a false rejection, never a write to the wrong term.

I'm not patching it directly onto this PR. The fix (discover the term directory from the export files instead of the Term) changes the paths that feed the file writes, the per-term PR, and the board move — so confirming it needs an end-to-end run, not just unit tests. That e2e runs from the dev fork's default branch, so it goes through our dev-first flow.

Merging as-is; hardening tracked in https://github.com/cncf/mentoring/issues/1938.
